eventsourcing/src/main/java/org/axonframework/eventsourcing/DefaultSnapshotterSpanFactory.java

```diff
@@ -0,0 +1,141 @@
+package org.axonframework.eventsourcing;
+
+import org.axonframework.common.BuilderUtils;
+import org.axonframework.tracing.Span;
+import org.axonframework.tracing.SpanFactory;
+
+import java.util.function.Supplier;
+
+import static java.lang.String.format;
+
+/**
+ * Default implementation of the {@link SnapshotterSpanFactory}. Can be configured to include the aggregate type in the
+ * span name (true by default) and to create a separate trace for the creation of the snapshot (false by default).
+ *
+ * @author Mitchell Herrijgers
+ * @since 4.9.0
+ */
+public class DefaultSnapshotterSpanFactory implements SnapshotterSpanFactory {
+    private final SpanFactory spanFactory;
+    private final boolean separateTrace;
+    private final boolean aggregateTypeInSpanName;
+
+    /**
+     * Creates a new {@link DefaultSnapshotterSpanFactory} using the provided {@code builder}.
+     *
+     * @param builder The builder to build the {@link DefaultSnapshotterSpanFactory} from.
+     */
+    protected DefaultSnapshotterSpanFactory(Builder builder) {
+        this.spanFactory = builder.builderSpanFactory;
+        this.separateTrace = builder.builderSeparateTrace;
+        this.aggregateTypeInSpanName = builder.builderAggregateTypeInSpanName;
+    }
+
+    @Override
+    public Span createScheduleSnapshotSpan(String aggregateType, String aggregateIdentifier) {
+        return spanFactory
+                .createInternalSpan(() -> createSpanName("scheduleSnapshot", aggregateType))
```

Comment:
Curious: Why change from a root span to an internal span compared to the previous setup?

Reply:
I'm not biased, I was just curious why you made the change is all.
If it's more inline with tracing defaults, I'm fine with making the change.

Although you could debate this _sudden_ change in approach is a breaking change from the perspective of assumptions.
I think we can let it fly this time, but I do think it's good to always be careful with behavior changes just as well.

Reply:
Alright noted. I understand that we need to be backwards compatible in our APIs. But, the changes in this release are to enhance the monitoring to reflect more of reality based on reality. I think the tracing API, as it's an insight tool, should not be treated as solidly as we do with the Java API. Otherwise we couldn't change the names of the traces or correct the way event processors work there as well.